packages/components/src/InlineMessage/InlineMessage.component.js

```diff
@@ -0,0 +1,86 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import Icon from './../Icon/Icon.component';
+import InlineMessageCss from './InlineMessage.scss';
+import { getTheme } from '../theme';
+
+/**
+ * @param {object} props react props
+ * @example
+ const props = {
+			type: 'Successful',
+			title: 'Successful',
+			description: 'Description',
+			icon: 'fa fa-check',
+			link: {
+				label: 'See more',
+				href: 'https://custom.url',
+			}
+			withBackground,
+	};
+ <InlineMessage {...props}/>
+ */
+
+export const TYPES = {
+	INFO: 'info',
+	SUCCESSFUL: 'success',
+	WARNING: 'warning',
+	ERROR: 'error',
+};
+
+export const getbsStyleFromType = type => {
+	const typesMap = {
+		[TYPES.INFO]: 'info',
+		[TYPES.SUCCESSFUL]: 'success',
+		[TYPES.WARNING]: 'warning',
+		[TYPES.ERROR]: 'danger',
+	};
+	return typesMap[type] || '';
+};
+
+const theme = getTheme(InlineMessageCss);
+
+export function InlineMessage({ type, title, description, icon, link, withBackground }) {
+	const rootClassnames = theme('tc-inline-message');
+
+	const textClasses = theme('tc-inline-message-text', getbsStyleFromType(type), {
+		background: withBackground,
+	});
+
+	const iconClassnames = theme('tc-inline-message-icon', getbsStyleFromType(type));
+
+	return (
+		<div className={rootClassnames}>
+			<span className={iconClassnames}>
+				<Icon name={icon} />
+			</span>
+			<div className={textClasses}>
```

Comment:
Would render an empty `<div>` if none of title/description/link is provided (might not happen in the real world though)

Reply:
Icon is mandatory, at least, so no it should not happen 🤷

Reply:
Yep it would in such a case. But not sure that we need extra check on title || description || link here, such as it's useless to have only icon when using this component